sched/clustermembership/cluster.go

```diff
@@ -3,8 +3,37 @@ package cluster_membership
 //go:generate mockgen -source=cluster.go -package=cluster_membership -destination=cluster_mock.go
 
 type Cluster interface {
-	/*
-	 * Returns a Slice of Node Ids in the Cluster
-	 */
+	//
+	// Returns a Slice of Node in the Cluster
+	//
 	Members() []Node
 }
+
+type DynamicCluster interface {
+	Cluster
+
+	// Returns a Channel which contains Node Updates
+	NodeUpdates() <-chan NodeUpdate
```

Comment:
Oh, I think I see now, because you're maintaining the nodes in the scheduler as elements in a channel, so you can't just get all the nodes afresh each time.